designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/ScenarioCommand.scala

```diff
@@ -0,0 +1,37 @@
+package pl.touk.nussknacker.ui.process.deployment
+
+import pl.touk.nussknacker.engine.api.deployment.{DMScenarioCommand, ScenarioActionName}
+import pl.touk.nussknacker.engine.api.process.ProcessIdWithName
+import pl.touk.nussknacker.engine.deployment.{CustomActionResult, ExternalDeploymentId}
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+
+import scala.concurrent.Future
+
+sealed trait ScenarioCommand[Result] {
+  def user: LoggedUser
+
+  implicit def implicitUser: LoggedUser = user
+}
+
+// Inner Future in result allows to wait for deployment finish, while outer handles validation
+// We split deploy process that way because we want to be able to split FE logic into two phases:
+// - validations - it is quick part, the result will be displayed on deploy modal
+// - deployment on engine side - it is longer part, the result will be shown as a notification
+case class RunDeploymentCommand(
+    processId: ProcessIdWithName,
+    savepointPath: Option[String],
+    comment: Option[String],
+    user: LoggedUser
+) extends ScenarioCommand[Future[Option[ExternalDeploymentId]]]
+
+case class CustomActionCommand(
```

Comment:
Is it still custom action? Why not just ActionCommand?

Reply:
As you can see, we still have built-in actions (run deployment, cancel scenario), so we need some word to distinguish between them.